### PR TITLE
Optional clear values

### PR DESCRIPTION
**Pull request by kvark**

### CODE REVIEW

**Comment by kainino0x** [comment]
Dictionary members are always optional by default if they don't have `required`, so the `?`s are not needed.

File: spec/index.bs [hunk 1]
```diff
@@ -639,19 +634,17 @@ dictionary GPURenderPassColorAttachmentDescriptor {
 
     required GPULoadOp loadOp;
```
Comment (kainino0x) [inline spec/index.bs]: Should also be removed?

Comment (kvark) [reply spec/index.bs]: nicely spotted!

**Comment by kvark** [comment]
@kainino0x thanks for the quick review! PTAL

**Comment by kainino0x** [comment]
But hey, it translates into a beautiful ADT in rust :)

**Comment by kvark** [comment]
@jdashg I'm curious what looks gross to you. Is it the optionals being simple types like `u32` and `float`, or something else?

**Comment by kainino0x** [comment]
I feel the same way a tiny bit, because it's not immediately obvious that null or unspecified means "load". Null/unspecified sounds like a "nothing" operation whereas "load" is not.

What if instead it were:
```webidl
enum GPULoadOp { "load" };
...
(GPULoadOp or GPUColor) loadValue;
```
? Plus it leaves the enum open for future extension.

**Comment by kvark** [comment]
> Null/unspecified sounds like a "nothing" operation whereas "load" is not.

I don't see this being the case because there is no longer any "operation" involved. If the user doesn't clear, they don't expect anything to happen (to that texture).

**Comment by kainino0x** [comment]
Sort of true, but in actual hardware and modern API concepts, the load isn't considered to be "nothing" either. I wouldn't want our mental model to move away from the native APIs.

**Comment by kvark** [comment]
I understand the concern. The pit of success is in encouraging the user to use clears where possible.
For example, the user draws a full-screen quad into a target. They don't particularly care about clearing, but doing so would be faster on a tiling GPU. Having the clear color as optional doesn't particularly encourage this.

In terms of API, ideally we want to express this type:
```rust
enum LoadOp {
  Load,
  Clear(Color),
}
```
Either the PR as it stands, or Kai's [comment](https://github.com/gpuweb/gpuweb/pull/276#issuecomment-487182659) would at least make this non-redundant. I'm fine with either.

**Comment by grorg** [comment]
Discussed at the [6 May 2019 teleconference](https://docs.google.com/document/d/1Lihb8Az6_xaizjAIQsu8cg0IEDBFOChTZGusd9D_ABE).

**Comment by kvark** [comment]
Agreement reached to proceed with Kai's [suggestion](https://github.com/gpuweb/gpuweb/pull/276#issuecomment-487182659).

**Pull request closed without merging**